build_runner/lib/src/asset_graph/graph.dart

```diff
@@ -164,10 +164,9 @@ class AssetGraph {
       Set<AssetId> newSources, String rootPackage) {
     newSources.map((s) => new AssetNode(s)).forEach(_add);
     var allInputs = new Set<AssetId>.from(newSources);
-    var phaseNumber = 0;
-    for (var action in buildActions) {
-      phaseNumber++;
+    for (var phase = 0; phase < buildActions.length; phase++) {
```

Comment:
could we break some of these changes out into separate commits and get them in early?